com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -70,12 +71,13 @@ public NetworkPrefabHandler PrefabHandler
             }
         }
 
-        /// <summary>
-        /// A synchronized time, represents the time in seconds since the server application started. Is replicated across all clients
-        /// </summary>
-        public float NetworkTime => Time.unscaledTime + m_CurrentNetworkTimeOffset;
+        public NetworkTimeSystem NetworkTimeSystem { get; private set; }
+
+        public NetworkTime PredictedTime => NetworkTimeSystem?.PredictedTime ?? new NetworkTime();
```

Comment:
I feel really silly for not using this pattern in my stuff.  So much cleaner, nice.